src/main/java/manager/DishManager.java

```diff
@@ -0,0 +1,88 @@
+package manager;
+
+import common.Messages;
+import entity.Dish;
+import ui.TextUi;
+import utils.DishStorage;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class DishManager {
+    private static ArrayList<Dish> dishes = new ArrayList<Dish>();
+
+    public DishManager(ArrayList<Dish> dishes) {
+        DishManager.dishes = dishes;
+    }
+
+    public static int getDishesSize() {
+        return dishes.size();
+    }
+
+    public static void addDishCommand(String name, int price, ArrayList<String> ingredients, TextUi ui) {
+        Dish dish = new Dish(name, price, ingredients);
+        dishes.add(dish);
+        try {
+            DishStorage dishStorage = new DishStorage();
+            dishStorage.writeToDishFile(dishes);
+        } catch (IOException e) {
+            ui.printMessage(String.format(Messages.ERROR_STORAGE_INVALID_WRITE_LINE, dish));
+        }
+    }
+
+    public static void deleteDishCommand(int index, TextUi ui) {
+        Dish selectedDish = dishes.get(index);
+        dishes.remove(index);
+        try {
+            DishStorage dishStorage = new DishStorage();
+            dishStorage.writeToDishFile(dishes);
+        } catch (IOException e) {
+            ui.printMessage(Messages.ERROR_STORAGE_DELETE_FAILED);
+        }
+    }
+
+    public static String viewDishCommand() {
+        int index = 1;
+        String everyDishInList = "";
+        for (Dish dish : dishes) {
+            everyDishInList += stringOfDishWithIndex(index, dish) + System.lineSeparator();
+            index++;
+        }
+        return everyDishInList;
+    }
+
+    public static String findDishCommand(String stringToFind) {
+        ArrayList<Dish> dishesMatchingKeyword = new ArrayList<>();
+        ArrayList<Integer> indexes = new ArrayList<Integer>();
+        for (int i = 0; i < getDishesSize(); i++) {
+            String[] words = dishes.get(i).getDishName().split(" ");
+            for (String word : words) {
+                if (word.equals(stringToFind)) {
+                    dishesMatchingKeyword.add(dishes.get(i));
+                    indexes.add(i + 1);
+                }
+            }
+        }
+
+        String dishesWithStringToFindInList = "";
```

Comment:
@Zeno-Zr 
This attribute's name is quite long, same with some of the other long names in this class